tempesta_fw/str.c

```diff
@@ -59,36 +59,42 @@ tfw_str_del_chunk(TfwStr *str, int id)
 static TfwStr *
 __str_grow_tree(TfwPool *pool, TfwStr *str, unsigned int flag, int n)
 {
+	TfwStr *a = NULL;
+
+	TFW_DBG("str_grow_tree:start:f:%d;f:%d;d:%s;\n", str->flags, flag,
+		str->data);
 	if (str->flags & flag) {
 		unsigned int l;
 		void *p;
 
+
 		if (unlikely(TFW_STR_CHUNKN_LIM(str))) {
 			TFW_WARN("Reaching chunks hard limit\n");
 			return NULL;
 		}
 
 		l = TFW_STR_CHUNKN(str) * sizeof(TfwStr);
-		p = tfw_pool_realloc(pool, str->ptr, l,
+		p = tfw_pool_realloc(pool, str->chunks, l,
 				     l + n * sizeof(TfwStr));
 		if (!p)
 			return NULL;
-		str->ptr = p;
+		str->chunks = p;
+		TFW_DBG("str_grow_tree:reall:%s\n", str->data);
 		TFW_STR_CHUNKN_ADD(str, n);
 	}
 	else {
-		TfwStr *a = tfw_pool_alloc(pool, (n + 1) * sizeof(TfwStr));
+		a = (TfwStr *)tfw_pool_alloc(pool, (n + 1) * sizeof(TfwStr));
 		if (!a)
 			return NULL;
 		a[0] = *str;
-		str->ptr = a;
+		str->chunks = (struct TfwStr *)a;
 		__TFW_STR_CHUNKN_SET(str, n + 1);
 	}
 
-	str = (TfwStr *)str->ptr + TFW_STR_CHUNKN(str) - n;
+	str = (TfwStr *)str->chunks + TFW_STR_CHUNKN(str) - n;
 	memset(str, 0, sizeof(TfwStr) * n);
 
-	return str;
+	return a;
```

Comment:
The change breaks logic of the function. Please leave the logic as it was before